File: buildserver/src/main/java/io/mohamed/rapid/buildserver/Result.java
package io.mohamed.buildserver;

import java.io.File;

public class Result {

  private final boolean successful;
  private File outputExtension;

  public Result(boolean successful, File outputExtension) {
    this.successful = successful;
    this.outputExtension = outputExtension;
  }

  public Result(boolean successful) {
    this.successful = successful;
  }

  public boolean isSuccessful() {
    return successful;
  }

  public File getOutputExtension() {
    return outputExtension;
  }
}

File: buildserver/src/main/java/io/mohamed/rapid/buildserver/Result.kt
package io.mohamed.rapid.buildserver

import java.io.File

class Result {
    val isSuccessful: Boolean
    var outputExtension: File? = null
        private set

    constructor(successful: Boolean, outputExtension: File?) {
        isSuccessful = successful
        this.outputExtension = outputExtension
    }

    constructor(successful: Boolean) {
        isSuccessful = successful
    }
}